core/tee/sub.mk

```diff
@@ -24,25 +24,29 @@ srcs-$(CFG_CRYPTO_CONCAT_KDF) += tee_cryp_concat_kdf.c
 srcs-$(CFG_CRYPTO_PBKDF2) += tee_cryp_pbkdf2.c
 
 ifeq ($(CFG_WITH_USER_TA),y)
-
+srcs-y += tee_obj.c
 srcs-y += tee_svc.c
-cppflags-tee_svc.c-y += -DTEE_IMPL_VERSION=$(TEE_IMPL_VERSION)
 srcs-y += tee_svc_cryp.c
 srcs-y += tee_svc_storage.c
-srcs-$(CFG_RPMB_FS) += tee_rpmb_fs.c
-srcs-$(CFG_REE_FS) += tee_ree_fs.c
-srcs-$(CFG_REE_FS) += fs_htree.c
-srcs-$(CFG_REE_FS) += fs_dirfile.c
-srcs-$(CFG_REE_FS) += tee_fs_rpc.c
-srcs-y += tee_fs_key_manager.c
-srcs-y += tee_obj.c
-srcs-y += tee_pobj.c
+cppflags-tee_svc.c-y += -DTEE_IMPL_VERSION=$(TEE_IMPL_VERSION)
 srcs-y += tee_time_generic.c
 srcs-$(CFG_SECSTOR_TA) += tadb.c
 srcs-$(CFG_GP_SOCKETS) += socket.c
 srcs-y += tee_ta_enc_manager.c
-
 endif #CFG_WITH_USER_TA,y
 
+ifeq ($(_CFG_WITH_SECURE_STORAGE),y)
+srcs-y += tee_fs_key_manager.c
+srcs-$(CFG_RPMB_FS) += tee_rpmb_fs.c
```

Comment:
Given that `_CFG_WITH_SECURE_STORAGE` will be `y` if `CFG_RPMB_FS` is `y` it seems unnecessary to have this inside the if.

Reply:
Correct, I will simplify.